Documentation/ApiOverview/Fluid/Index.rst

```diff
@@ -0,0 +1,214 @@
+
+.. sidebar:: Sitepackage
+
+   In TYPO3, (almost) everything is an extension. If you want to create a theme, override
+   some configuration or add custom content elements you will do this in a custom TYPO3
+   extension. A sitepackage is an extension that contains everything that is necessary
+   for your site: The configuration, necessary assets for your theme (images, css etc.)
+   and a combination of TypoScript and Fluid.
+
+You can use Fluid in TYPO3 to do one of the following:
+
+* Create a template (theme) in combination with TypoScript :ref:`FLUIDTEMPLATE <t3tsref:t3tsrefcobj-fluidtemplate>`.
+  Check out the :ref:`t3sitepackage:start` which walks you through the
+  creation of a sitepackage extension.
+* :ref:`adding-your-own-content-elements` in addition to the already existing content
+  elements TYPO3 supplies.
+* The the previous point describes the lightweight components which
+  are created using a combination of TypoScript and Fluid. If you need more functionality
+  or flexibility in your content element, you can create a content plugin using
+  a combination of Extbase and Fluid. This is explained in :ref:`t3extbasebook:start`
+* Use Fluid to create emails using the :ref:`TYPO3 Mail API <mail-fluid-email>`.
+
+Example Fluid snippet
+=====================
+
+This is how a simple Fluid snippet could look like::
+
+   <h4>This is your headline</h4>
+   <p>
+    <f:if condition="{myExpression}">
+      <f:then>
+         {somevariable}
+      </f:then>
+      <f:else>
+          {someothervariable}
+      </f:else>
+    </f:if>
+   </p>
+
+The resulting HTML may look like this::
+
+   <h4>This is your headline</h4>
+   <p>This is the content of variable "somevariable"</p>
+
+The above Fluid snippet contains:
+
+ViewHelpers:
+   The XML elements that start with `f:` like `<f:if>` etc. are ViewHelpers. These
+   are PHP components that are supplied by Fluid and can be used in your Fluid templates.
+   TYPO3 adds some more ViewHelpers for TYPO3 specific functionality. And, you can
+   :ref:`write your own <t3extbasebook:developing-a-custom-viewhelper>`.
+
+   ViewHelpers can do simple processing such as remove spaces with the
+   :ref:`t3viewhelper:typo3fluid-fluid-spaceless` ViewHelper or create a link
+   as is done in the TYPO3 Fluid ViewHelper :ref:`t3viewhelper:link`.
+
+Object Accessors:
+   Fluid can access variables that have been defined. Just use braces
+   and the name of the variable: `{somevariable}`. In Fluid, these placeholders
+   are called `Object Accessors`.
+
+Conditions:
+    The conditions are supplied here by the if / then / else ViewHelpers.
+
+Directory structure
+===================
+
+In your extension, the following directory structure should be used for Fluid files:
+
+.. code-block:: none
+
+   ── Resources
+      └── Private
+        ├── Layouts
+        ├── Partials
+        └── Templates
+
+How you structure the files within the respective `Layouts`, `Partials` and `Templates`
```

Comment:
We could add a note here saying something like "this folder structure is the standard in TYPO3 CMS. When using Fluid outside of TYPO3 CMS you can use any folder structure you like" - and perhaps we should also have a very brief section about how you can define these paths, either with PHP API (RenderingContext->TemplatePaths) or TS (the latter only if we want to make this landing page more TYPO3 CMS specific, of course).